custom_components/versatile_thermostat/underlyings.py

```diff
@@ -202,6 +204,7 @@ def __init__(
         self._on_time_sec = 0
         self._off_time_sec = 0
         self._hvac_mode = None
+        self._keep_alive = IntervalCaller(hass, keep_alive_sec)
```

Comment:
Note: To simplify the code, I chose to always instantiate this object even if `keep_alive_sec` is unset or zero, but in this case it will do nothing and the feature will be effectively disabled. I tested it when `keep_alive_sec` is unset, zero, and non-zero.